variables.tf

```diff
@@ -109,6 +109,12 @@ variable "use_azure_cni" {
   default     = false
 }
 
+variable "use_azure_cni_overlay" {
```

Comment:
Could we instead do something like:

```
variable "azure_cni" {
  type = object({
    enabled = optional(bool, false)
    overlay_mode = optional(bool, false)
  })
  validation {
    error_message = "Overlay mode can only be true if azure_cni is enabled"
    condition = XXXXX
  }
 ```
 
 I feel it makes more sense to add validation to expose to the end user that there is a problem with trying to enable `use_azure_cni_overlay` without first enabling `use_azure_cni`. Instead we are currently using `var.use_azure_cni && var.use_azure_cni_overlay ? "overlay" : null` within `main.tf` to fix the issue for the end user, however this may be unexpected behaviour that the module is doing without feeding back to the end user

Reply:
I was considering this but in the interest of keeping the config similar so there's no additional changes to account for.

If you're happy for me to implement this and update it across the board I'm happy to implement the change into a minor release.

Reply:
See updated value - resolving this one for now